output-comparator/src/main/java/org/mobilitydata/gtfsvalidator/outputcomparator/cli/Main.java

```diff
@@ -0,0 +1,127 @@
+      return;
+    }
+    if (reportDirectory.length == 0) {
+      logger.atSevere().log(
+          "Specified directory is empty, cannot generate acceptance tests report.");
+      return;
+    }
+    ImmutableMap.Builder<String, Integer> mapBuilder = new Builder<>();
+    int badDatasetCount = 0;
+    List<File> reportDirs =
+        Arrays.stream(reportDirectory).filter(File::isDirectory).collect(Collectors.toList());
+
+    int totalDatasetCount = reportDirs.size();
+
+    try {
+      for (File file : reportDirs) {
+        int newErrorCount = 0;
+        ValidationReport referenceReport =
+            ValidationReport.fromPath(
+                file.toPath().resolve(args.getReferenceValidationReportName()));
+        ValidationReport latestReport =
+            ValidationReport.fromPath(file.toPath().resolve(args.getLatestValidationReportName()));
+
+        if (referenceReport.hasSameErrorCodes(latestReport)) {
+          mapBuilder.put(file.getName(), newErrorCount);
+          continue;
+        }
+        if (referenceReport.equals(latestReport)) {
+          mapBuilder.put(file.getName(), newErrorCount);
+          continue;
+        }
+        newErrorCount = referenceReport.getNewErrorCount(latestReport);
+        mapBuilder.put(file.getName(), newErrorCount);
+        if (newErrorCount >= args.getThreshold()) {
+          badDatasetCount += 1;
+        }
+      }
+      exportAcceptanceReport(mapBuilder.build(), args.getReportDirectory());
+      isNewRuleValid(badDatasetCount, totalDatasetCount, args.getAcceptanceCriteria());
+    } catch (IOException e) {
+      logger.atSevere().withCause(e);
+      System.exit(1);
+    }
+  }
+
+  /**
+   * Exports the acceptance test report (map of String, Object) as json.
+   *
+   * @param acceptanceReportData acceptance report content.
+   * @param outputBase base path to output.
+   * @throws IOException if an I/O error occurs writing to or creating the file.
+   */
+  private static void exportAcceptanceReport(
+      ImmutableMap<String, Integer> acceptanceReportData, String outputBase) throws IOException {
+    Gson gson = new GsonBuilder().serializeNulls().create();
+    Files.write(
+        Paths.get(outputBase, ACCEPTANCE_REPORT_JSON),
+        gson.toJson(acceptanceReportData).getBytes(StandardCharsets.UTF_8));
+  }
+
+  /**
+   * Exits on non-zero code 2 if the ratio badDatasetCount/totalDatasetCount exceeds the threshold
+   * defined as acceptance criteria.
+   *
+   * @param badDatasetCount the number of nw invalid datasets
+   * @param totalDatasetCount the number of datasets to be tested
+   * @param threshold the acceptance criteria
+   */
+  private static void isNewRuleValid(int badDatasetCount, int totalDatasetCount, double threshold) {
+    System.out.printf(
+        "%d out of %d datasets (~%.2f %%)  are invalid due to new implementation.%n",
+        badDatasetCount, totalDatasetCount, 100.0 * badDatasetCount / totalDatasetCount);
```

Comment:
`100.0 * badDatasetCount / totalDatasetCount` is being computed multiple times here - I'd pull this out to a variable.